src/AddProjectsPage/index.tsx

```diff
@@ -61,76 +61,70 @@ class AddProjectsPage extends React.Component<
-          tags: project.tags,
-          images: project.images,
-          contact: project.contact,
-          creator: project.creator,
-          categoryPlaceholder: 'Choose A Category',
-          tagPlaceholder: 'Choose Some Tags',
-          teamPlaceholder: 'Add Teammates',
-          statusPlaceholder: 'Status of Project',
-          preview: null,
-          files: null
-        },
-        () => {
-          // console.log(this.state);
-          // var doc: any;
-          // if (this.state.lookingFor === ['Programmer']) {
-          //   doc = document.getElementById('new-project-role-p')!;
-          //   doc.checked = true;
-          // } else if (this.state.lookingFor === ['Designer']) {
-          //   doc = document.getElementById('new-project-role-d')![0];
-          //   doc.checked = true;
-          // } else if (this.state.lookingFor!.indexOf('Programmer') !== -1
-          //   && this.state.lookingFor!.indexOf('Designer') !== -1) {
-          //   var d: HTMLInputElement = document.getElementById('new-project-role-d')![0];
-          //   var p: HTMLInputElement = document.getElementById('new-project-role-p')![0];
-          //   d.checked = true;
-          //   p.checked = true;
-          // }
-        }
-      );
-    };
-
-    async function callProjectAssignToState() {
-      await getProjectById();
-      await setState();
-    }
-
     // if there was an ID passed in with the url link
-    // call data of that one project
-    // update state with requested project data
     if (this.props.match.params.hasOwnProperty('id')) {
-      id = this.getURLParams().id;
-      callProjectAssignToState();
+      // call data of that one project
+      this.props
+        .getOneProject(this.props.match.params.id)
+        // then update state with requested project data
+        .then(() => {
+          var project = this.props.currentProject!;
+          this.setState({
+            name: project.name,
+            description: project.description,
+            dueDate:
+              project.dueDate !== null ? project.dueDate.slice(0, 10) : '',
+            team: project.team,
+            githubLink: project.githubLink,
+            mockupLink: project.mockupLink,
+            liveLink: project.liveLink,
+            lookingFor: project.lookingFor,
+            status: project.status,
+            category: project.category,
+            tags: project.tags,
+            images: project.images,
+            contact: project.contact,
+            creator: project.creator,
+            categoryPlaceholder: 'Choose A Category',
+            tagPlaceholder: 'Choose Some Tags',
+            teamPlaceholder: 'Add Teammates',
+            statusPlaceholder: 'Status of Project',
+            preview: null,
+            files: null
+          });
+        })
+        // then toggles roles checkbox per project lookingFor array
+        .then(() => {
+          var p = document.getElementById(
```

Comment:
Instead of trying to do DOM manipulation here, we should let react render out the component with that state.